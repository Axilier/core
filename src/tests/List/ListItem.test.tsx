import React from 'react';
import { mount } from 'enzyme';
import { ListItem, Key } from '../../index';

const onClick = jest.fn();
const listItem = mount(
    <ListItem dropdown onClick={onClick}>
        test
    </ListItem>,
);
const cleanUp = () => listItem.setProps({ dropdown: true, onClick });

describe('when I render the List in an application', () => {
    it('should render correctly', () => {
        expect(listItem).toMatchSnapshot();
    });
    describe('when the `controls` are defined', () => {
        beforeAll(() =>
            listItem.setProps({
                controls: (
                    <>
                        <Key className={'controls'} />
                    </>
                ),
            }),
        );
        it('should render an extra fragment that is the contained controls', () => {
            expect(listItem.find('.controls').exists()).toBeTruthy();
        });
    });
    describe('when the listItem is not selected', () => {
        describe('when the listItem is being hovered over', () => {
            beforeAll(() => listItem.simulate('mouseenter'));
            it('should have a backgroundColor that is #F9F9F9', () => {
                expect(listItem.childAt(0).props().style).toHaveProperty(
                    'backgroundColor',
                    '#F4F4F4',
                );
            });
            afterAll(() => listItem.simulate('mouseleave'));
        });
        describe('when the listItem is not being hovered over', () => {
            it('should have a backgroundColor that is #F9F9F9', () => {
                expect(listItem.childAt(0).props().style).toHaveProperty(
                    'backgroundColor',
                    '#F9F9F9',
                );
            });
        });
    });
    describe('when the listItem is selected', () => {
        beforeAll(() => listItem.setProps({ selected: true }));
        it('should have a backgroundColor of #EDF3FD', () => {
            expect(listItem.childAt(0).props().style).toHaveProperty(
                'backgroundColor',
                '#EDF3FD',
            );
        });
        afterAll(cleanUp);
    });
    describe('when the `dropdown` prop is defined', () => {
        it('should render a dropdown arrow', () => {
            expect(listItem.find('.dropdown-test').exists()).toBeTruthy();
        });
        describe('when `dropdown` is set to true', () => {
            it('should render the dropdown arrow up', () => {
                expect(
                    listItem.find('.dropdown-test').props().style,
                ).toHaveProperty('transform', 'rotate(180deg)');
            });
        });
        describe('when `dropdown` is set to false', () => {
            beforeAll(() => listItem.setProps({ dropdown: false }));
            it('should render the dropdown arrow down', () => {
                expect(
                    listItem.find('.dropdown-test').props().style,
                ).toHaveProperty('transform', 'none');
            });
            afterAll(cleanUp);
        });
    });
    describe('when the `dropdown` prop is not defined', () => {
        beforeAll(() => listItem.setProps({ dropdown: undefined }));
        it('should not render the dropdown arrow', () => {
            expect(listItem.find('.dropdown-test').exists()).toBeFalsy();
        });
        afterAll(cleanUp);
    });
    describe('when the listItem is clicked', () => {
        describe('when `onClick` is defined', () => {
            beforeAll(() => listItem.simulate('click'));
            it('should run the provided callback', () => {
                expect(onClick).toHaveBeenCalled();
            });
        });
        describe('when `onClick` is not defined', () => {
            beforeAll(() => {
                listItem.setProps({ onClick: null });
                listItem.simulate('click');
            });
            it('should do nothing', () => {
                expect(true).toBeTruthy();
            });
        });
    });
    describe('when the `child` prop is true', () => {
        beforeAll(() => listItem.setProps({ child: true }));
        it('should render with paddingLeft of 40px', () => {
            expect(listItem.childAt(0).props().style).toHaveProperty(
                'paddingLeft',
                '40px',
            );
        });
        afterAll(cleanUp);
    });
    describe('when the `child` prop is false', () => {
        beforeAll(() => listItem.setProps({ child: false }));
        it('should render with paddingLeft of 30px', () => {
            expect(listItem.childAt(0).props().style).toHaveProperty(
                'paddingLeft',
                '30px',
            );
        });
        afterAll(cleanUp);
    });
});
