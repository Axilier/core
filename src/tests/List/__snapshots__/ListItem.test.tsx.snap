// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when I render the List in an application should render correctly 1`] = `
<ListItem
  dropdown={true}
  onClick={[MockFunction]}
>
  <Layout
    className="listItem"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    orientation="row"
    style={
      Object {
        "backgroundColor": "#F9F9F9",
        "paddingLeft": "30px",
      }
    }
  >
    <div
      className="listItem"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "backgroundColor": "#F9F9F9",
          "display": "flex",
          "flexDirection": "row",
          "paddingLeft": "30px",
        }
      }
    >
      <div
        className="listItemChildren"
      >
        test
      </div>
      <div
        className="endFragment"
      >
        <svg
          className="dropdown-test"
          fill="black"
          height="16"
          style={
            Object {
              "transform": "rotate(180deg)",
            }
          }
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.204 5h9.592L8 10.481 3.204 5zm-.753.659 4.796 5.48a1 1 0 0 0 1.506 0l4.796-5.48c.566-.647.106-1.659-.753-1.659H3.204a1 1 0 0 0-.753 1.659z"
          />
        </svg>
      </div>
    </div>
  </Layout>
</ListItem>
`;
